refector: Food 도메인 재구조화(#35, #40)

hdg5639 opened the pull request:
<!-- 
PR 제목 작성 가이드:
형식: [타입]: 간단한 설명 (#이슈번호)

타입 예시:
- feat: 새로운 기능 추가
- fix: 버그 수정  
- docs: 문서 수정
- style: 코드 포맷팅
- refactor: 코드 리팩토링
- test: 테스트 추가/수정
- chore: 빌드, 패키지 등 기타 작업

예시: feat: 사용자 로그인 API 구현 (#123)
-->

## 📝 작업 내용
<!-- 이번 PR에서 구현한 기능이나 수정한 내용을 구체적으로 작성해주세요 -->
- Food 관련 도메인(관리자용 메뉴 생성/관리 시스템) 전면 재구조화
- Entity 네이밍 변경: FoodMenu → Food, MealOption → FoodChoiceSet, WeeklyMealPlan → WeeklyFoodPlan 등
- Repository 인터페이스 및 쿼리 메서드명 정리 및 테스트
- Service 계층 재구조화: FoodService / WeeklyFoodPlanService로 분리
- DTO 클래스 네이밍 일괄 변경
- DAO 삭제
- Controller 및 API 경로 재설계
- Swagger 문서 반영

## 🔗 관련 이슈
<!-- 관련된 이슈가 있다면 링크해주세요 -->
- Related to #35 , #40 

## 💬 추가 요청사항
<!-- 리뷰어에게 특별히 확인받고 싶은 부분이나 질문이 있다면 작성해주세요 --> 
이전 Meal 관련 테이블 전체 삭제 필요

## ✅ 체크리스트
### 코드 품질
- [x] 커밋 컨벤션 준수 (feat/fix/docs/refactor 등)
- [x] 불필요한 코드/주석 제거

### 테스트
- [x] 로컬 환경에서 동작 확인 완료
- [x] 기존 기능에 영향 없음 확인

### 리뷰
- [x] 적절한 리뷰어 지정 완료
- [ ] 리뷰어 1명 이상 승인

### 배포 준비
- [x] 환경변수 추가/변경사항 문서화
- [x] DB 마이그레이션 필요 여부 확인
- [ ] 배포 시 주의사항 없음

hdg5639 committed:
refactor: Food 도메인 재구조화

- Food 관련 도메인(관리자용 메뉴 생성/관리 시스템) 전면 재구조화
- Entity 네이밍 변경: FoodMenu → Food, MealOption → FoodChoiceSet, WeeklyMealPlan → WeeklyFoodPlan 등
- Repository 인터페이스 및 쿼리 메서드명 정리 및 테스트
- Service 계층 재구조화: FoodService / WeeklyFoodPlanService로 분리
- DTO 클래스 네이밍 일괄 변경
- Controller 및 API 경로 재설계
- Swagger 문서 반영

Issue: #35

hdg5639 committed:
refactor: Food 관련 DAO 삭제
- 사용하지 않는 DAO 레이어 삭제

Issue: #35
Related: #40

hdg5639 committed:
refactor: Food Repository 패키지명 변경
- meal -> food로 패키지명 변경

Issue: #35

ryusemin reviewed (dismissed):
확인하였습니다.

2-jjong commented on Backend/src/main/java/com/demo/nimn/dto/food/FoodChoiceSetDTO.java:
foodType이 아침, 점심, 저녁 식사를 나타내는 것인데 변수명이 명확하지 않아서 혼동이 있을 것 같습니다. enum으로 분리하거나 적절한 변수명으로 수정 제안드립니다.

2-jjong commented on Backend/src/main/java/com/demo/nimn/entity/food/FoodChoiceSet.java:
FoodChoiceSetDTO는 foodType FoodChoiceSet은 mealType
DTO와 Entity의 변수명이 달라 혼동이 있을 수 있습니다. DTO에 달린 코멘트 보시고 적절한 변수명으로 통일해주세요.

2-jjong commented on Backend/src/main/java/com/demo/nimn/service/food/FoodServiceImpl.java:
```diff
@@ -0,0 +1,138 @@
+        return toFoodDTOS(foods);
+    }
+
+    public List<FoodDTO> toFoodDTOS(List<Food> foods) {
```
entity와 dto 변환 메소드명을 통일해주세요.
DTO: convertToXXXDTO
Entity: convertToXXX

2-jjong replied:
다른 클래스에도 똑같이 수정해주세요.

2-jjong commented on Backend/src/main/java/com/demo/nimn/entity/food/Food.java:
```diff
@@ -0,0 +1,64 @@
+    @OneToOne(mappedBy = "food", cascade = CascadeType.ALL)
+    private ReviewSummary reviewSummary;
+
+    public FoodDTO toFoodDTO() {
```
다른 Entity와 동일하게 DTO 변환 메소드를 Entity 내부가 아닌 Service 레이어로 변경해주세요.

2-jjong commented on Backend/src/main/java/com/demo/nimn/service/food/WeeklyFoodPlanServiceImpl.java:
이전에는 Unique한 Food를 선택지로 사용하도록 했는데 한 플랜에 같은 메뉴가 중복될 수 있습니다. 즉, 아침, 점심, 저녁 중에 아침, 점심의 선택지 3개 중에 같은 메뉴가 있을 수 있습니다. 따라서 Unique한 Food 선택이 아닌 동일한 랜덤값으로 선택할 수 있도록 수정해주세요.

2-jjong commented on Backend/src/main/java/com/demo/nimn/controller/food/FoodController.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Operation(summary = "한 주 식단 생성(CreateWeeklyFoodPlan)", description = "한 주 식단 생성")
+    @PostMapping(value = "/food-plans/weekly")
+    public WeeklyFoodPlanDTO createWeeklyFoodPlan() {
```
한 주 Plan을 직접 API로 요청되어 생성하게 되면 프론트에서 따로 스케줄링이나 추가 작업이 필요합니다.
백엔드단에서 스케줄링을 사용하거나 한 주 식단을 조회했을 때 해당 주에 식단이 없다면 그때 생성해서 사용하는 방안이 더 적절한 것 같은데 어떻게 생각하세요?

2-jjong commented on Backend/src/main/java/com/demo/nimn/controller/food/FoodController.java:
Food -> 메뉴, 음식
Diet -> Plan에서 사용자가 최종 선택한 식단
Plan -> 식단을 선택하기 위한 선택지 모음

Swagger를 작성할 때 용어의 혼동이 있지 않도록 적절하게 수정바랍니다.

2-jjong reviewed (changes requested):
코멘트 읽어보시고 적절히 수정해주세요.

hdg5639 committed:
refactor: FoodType 열거형 분리 및 변수명 통일, Entity <-> DTO 변환 메소드명 통일 및 스케줄러 도입
- FoodType 열거형으로 분리 및 FoodChoiceSet 과 DTO 변수명 통일
- Entity <-> DTO 변환 메소드명 통일 및 위치 수정
- 한 주 식단 생성 기능에 스케쥴러 도입, 수동 생성과 병행하여 사용 가능
- 식단 생성 시 존재하던 중복 방지, 가중치 로직 제거 및 완전 랜덤으로 변경
- Swagger 문서 수정

Issue: #35

hdg5639 committed:
refactor: FoodController 엔드포인트 수정
- FoodController의 엔드포인트를 좀 더 직관적으로 수정

Issue: #35

hdg5639 committed:
refactor: Swagger 문서 개선 및 FoodType -> FoodTime 열거형 이름 수정
- Swagger 문서 개선
- 모호한 FoodType 열거형을 FoodTime 으로 수정

Issue: #35
